doc/src/manual/command-line-interface.md

```diff
@@ -46,8 +46,29 @@ but at the conclusion of executing a script or expression, `julia` will attempt
 `Main.main(Base.ARGS)` if such a function `Main.main` has been defined and this behavior was opted into
 by using the `@main` macro.
 
-This feature is intended to aid in the unification
-of compiled and interactive workflows. In compiled workflows, loading the code that defines the `main`
+To see this feature in action, consider the following definition:
+```jldoctest
+function (@main)(ARGS)
+    open("out.log", "w") do io
+        println(io, "My pet is $(ARGS[1])")
+    end
+    return nothing
+end
+```
+Executing the above script with `julia script.jl "Buddy"` will automatically run `(@main)` and create the `out.log`
+file, despite there being no explicit call to `(@main)`.
+
+The return value of the `(@main)` function is the exit code, and must be a value convertible to `Cint`
+(`nothing` will be replaced by `Cint(0)`):
```

Comment:
```suggestion
The return value of the `(@main)` function is the exit code, and must be a value convertible to `Cint`
(as a special case, `nothing` is converted to `Cint(0)`):
```

Reply:
Of course. Hence "as a sepcial case" and the use of "converted" rather than "`convert`ed". I'm open to other suggestions.

Reply:
proposal:

```md
(as a special case, a return value of `nothing` is treated as if `Cint(0)` was returned):
```

Reply:
The part that I don't like is "must be a value convertible to `Cint`". My proposal was to adjust the definition of "convertible" in that sentence with a special case, though I agree it is not ideal.

Either of these are better:

```suggestion
The return value of the `(@main)` function is the exit code, and must either be `nothing`
(exit code `0`) or be a value convertible to `Cint`
```

```suggestion
The return value of the `(@main)` function must either be `nothing`, resulting in exit code
`0`, or convertible to a `Cint` which will be the exit code:
```